include/RAJA/internal/Span.hpp

```diff
@@ -88,18 +88,18 @@ struct Span {
                 "ValueType must model RandomAccessIterator");
 
   iterator begin() { return _data; }
-  iterator end() { return std::advance(_data, _length); }
+  iterator end() { return _data + _length; }
   const_iterator begin() const { return _data; }
-  const_iterator end() const { return std::advance(_data, _length); }
+  const_iterator end() const { return _data + _length; }
```

Comment:
Uh... That's scary.  Is this because they aren't always iterators, or is there something going on here that we might want to look into?